jadx-plugins/jadx-plugins-api/src/main/java/jadx/api/plugins/utils/ZipSecurity.java

```diff
@@ -73,4 +78,28 @@ public static boolean isValidZipEntry(ZipEntry entry) {
 		return isValidZipEntryName(entry.getName())
 				&& !isZipBomb(entry);
 	}
+
+	public static InputStream getInputStreamForEntry(ZipFile zipFile, ZipEntry entry) throws IOException {
+		InputStream in = zipFile.getInputStream(entry);
+		LimitedInputStream limited = new LimitedInputStream(in, entry.getSize());
+		return new BufferedInputStream(limited);
+	}
+
+	public static void visitZipEntries(File file, BiConsumer<ZipEntry, InputStream> visitor) {
+		try (ZipFile zip = new ZipFile(file)) {
+			zip.stream()
+					.filter(entry -> !entry.isDirectory())
+					.filter(ZipSecurity::isValidZipEntry)
+					.limit(MAX_ENTRIES_COUNT)
```

Comment:
Using the streams `limit` functionality is very clean, but if we hit the limit it will be not noticeable by the user.
Therefore in this case a manual counter + Exception if the limit is hit may be better.

Reply:
I agree, looks like it is better to rewrite with a simple loop